test_ak_survey_results.py

```diff
@@ -1,16 +1,40 @@
 import pytest
 import test_settings
-from ak_survey_results import AKSurveyResults
+import boto3
+from moto import mock_secretsmanager
 from ak_survey_results import PageNotFoundException
 from ak_survey_results import PageNotSurveyException
 from ak_survey_results import PageNotLoadedException
 from ak_survey_results import Struct
 
 
+mock_secret = {'DB_SCHEMA_AK': 'ak', 'DB_SCHEMA_SURVEY': 'survey_results', 'DB_TYPE': 'PostgreSQL', 'COLUMN_EXCLUDES': ''}
+redshift_secret={'username':'mock_admin_username','password':'mock_admin_password','host':'localhost','port':'5432'}
+
+class ArgsObject:
+    def __init__(self, dict_args):
+        for arg, val in dict_args:
+            setattr(self, arg, val)
+
+@mock_secretsmanager
+def init_secrets():
+    client = boto3.client('secretsmanager')
+    client.create_secret(Name='ak-survey-results', SecretString=json.dumps(mock_secret))
+    client.create_secret(Name='redshift-admin', SecretString=json.dumps(redshift_secret))
+
+
+@mock_secretsmanager
 class Test:
 
     def setup(self):
```

Comment:
Thanks for updating these tests! Unfortunately, I'm getting a really weird error when I try to run them: `TypeError: setup() takes 1 positional argument but 2 were given`. Were you able to run the tests without this error?

I can't actually find any pytest documentation on the `setup` function (as opposed to `setup_module`, `setup_class`, etc. [as described here](https://docs.pytest.org/en/6.2.x/xunit_setup.html)), so I'm wondering if updating to the latest pytest broke something for me unrelated to these changes. What pytest version do you have?

When I add a second parameter to the function and then set a breakpoint, I can see that the test function is being passed to the setup function as the second parameter. I'm not sure this is what we want, because it seems to imply the setup function is being run for a specific test, when I think we only want it to run once, before all tests.

Reply:
Thanks for your research into this!  I think I've fixed all the remaining issues - I was able to run the tests locally by inputting the appropriate credentials into the mock secrets.